src/types/types.tsx

```diff
@@ -6,27 +8,40 @@ export type Vendor = {
   name: string;
 };
 
+export type VoucherRange = {
+  startSerialNum: number;
+  endSerialNum: number;
+  type: string;
+  maxValue: number;
+};
+
 export type Voucher = {
-  uuid: Uuid;
-  type: VoucherType;
+  serialNumber: number;
+  type: string;
   value: number;
   vendorUuid: Uuid;
-  expirationDate: string;
-  status: VoucherStatus;
 };
 
 export type VoucherCreate = Pick<
   Voucher,
-  'type' | 'value' | 'vendorUuid' | 'expirationDate' | 'status'
+  'serialNumber' | 'vendorUuid' | 'value'
 >;
 
-export enum VoucherType {
-  GREEN = 'green',
-  ORANGE = 'orange',
-  PURPLE = 'purple',
-}
+export type Transaction = {
+  uuid: Uuid;
+  timestamp: Timestamp;
+  status: TransactionStatus;
+  value: number;
+  voucherArray: number[];
```

Comment:
Consider renaming to something to indicate that this is an array of voucher serial numbers bc otherwise this could be confused for an array of Voucher objects.

Something like `voucherSerialNumbers`?

Reply:
This applies to most places you use `voucherArray`